moq-relay: init at v0.6.2

therishidesai opened the pull request:
## Description of changes

Create nixos module for moq-relay. moq-relay package comes from #342588
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

therishidesai committed:
moq-relay: init at v0.6.2

therishidesai committed:
nixos/moq-relay: init moq-relay systemd service

s0me1newithhand7s commented:
pls, try to not ping 3500+ ppls this time. 💀

hraban commented:
> pls, try to not ping 3500+ ppls this time. 💀

🤷 as one of the people getting pinged, at what point do we take responsibility for the fragility of this system? do we truly expect nobody to ever make the mistake of merging off staging and accidentally creating a PR into the _default_ base branch offered by GH (master)? 😛 I know I've been there.

thanks for the contribution  @therishidesai  :)

therishidesai commented:
> pls, try to not ping 3500+ ppls this time. 💀

Sorry about that

nyabinary commented:
Can't you just have one PR and init the package in one commit and init the module in the other 2 commits?

nyabinary commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5828,6 +5828,10 @@ with pkgs;
 
   mousetweaks = callPackage ../applications/accessibility/mousetweaks { };
 
+  moq-relay = callPackage ../by-name/mo/moq-relay/package.nix {
+    inherit libressl pkg-config;
+  };
+
```
IIRC, this isn't required for packages in ``by-name``.

therishidesai replied:
fixed

nyabinary reviewed (changes requested):
IIRC this isn't required for packages in ``by-name``.

dotlambda commented on pkgs/by-name/mo/moq-relay/package.nix:
```diff
@@ -0,0 +1,35 @@
+
+  buildInputs = [ libressl ];
+
+  meta = with lib; {
```
```suggestion
  meta = {
```
#208242

dotlambda commented on nixos/modules/services/video/moq-relay.nix:
```diff
@@ -0,0 +1,46 @@
+    systemd.services.moq-relay = {
+      description = "Media over QUIC relay server";
+      path = [ pkgs.moq-relay ];
+      script = ''
```
use `ExecStart` instead

dotlambda reviewed (commented):
Commit message should be `moq-relay: init at 0.6.0`.
Please add some hardening to the service.

therishidesai commented:
> Can't you just have one PR and init the package in one commit and init the module in the other 2 commits?

Was just basing this off of other packages where the package was added in an init PR and the nixos module was in another PR

therishidesai commented:
> Commit message should be `moq-relay: init at 0.6.0`. Please add some hardening to the service.

I had a separate PR to just add the `moq-relay` package. Is it fine that this PR has 2 commits? In the past I remember maintainers asking for single atomic commits in the PR

therishidesai commented:
> Please add some hardening to the service.

I'm not too familiar with the systemd hardening. Are there any services I should look at for hardening examples? What specifically makes sense for this server?

dotlambda commented:
> Are there any services I should look at for hardening examples?

Just grep for "hardening" in `nixos/`.

therishidesai commented:
> > Are there any services I should look at for hardening examples?
> 
> Just grep for "hardening" in `nixos/`. Also see https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html.

Added some new hardening and cleaned up users/groups similar to other server modules in nixos.
